### PR TITLE
Unit Script files Tidy up Part 4

### PR DESCRIPTION
**Pull request by MrRowey**

### CODE REVIEW

File: units/UEL0001/UEL0001_script.lua [hunk 1]
```diff
@@ -134,22 +131,22 @@ UEL0001 = ClassUnit(ACUUnit) {
         if rebuildDrone == true then
             if pod == 'LeftPod' then
                 if self.HasLeftPod == true then
-                    self.RebuildThread = self:ForkThread(self.RebuildPod, 1)
+                    self.RebuildThread = self.Trash:Add(ForkThread(self.RebuildPod, 1,self))
```
Comment (Garanas) [inline units/UEL0001/UEL0001_script.lua:134]: Not a valid replacement, the `Add` function doesn't return the thread

File: units/UEL0301/UEL0301_script.lua [hunk 2]
```diff
@@ -76,7 +74,7 @@ UEL0301 = ClassUnit(CommandUnit) {
     NotifyOfPodDeath = function(self, pod, rebuildDrone)
         if rebuildDrone == true then
             if self.HasPod == true then
-                self.RebuildThread = self:ForkThread(self.RebuildPod)
+                self.RebuildThread = self.Trash:Add(ForkThread(self.RebuildPod,self))
```
Comment (Garanas) [inline units/UEL0301/UEL0301_script.lua:77]: Not a valid replacement

File: units/UEL0307/UEL0307_script.lua [hunk 3]
```diff
@@ -63,7 +55,7 @@ UEL0307 = ClassUnit(TShieldLandUnit, ShieldEffectsComponent) {
         TShieldLandUnit.OnShieldDisabled(self)
         ShieldEffectsComponent.OnShieldDisabled(self)
         KillThread(self.DestroyManipulatorsThread)
-        self.DestroyManipulatorsThread = self:ForkThread(self.DestroyManipulators)
+        self.DestroyManipulatorsThread = self.Trash:Add(ForkThread(self.DestroyManipulators, self))
```
Comment (Garanas) [inline units/UEL0307/UEL0307_script.lua:58]: Not a valid replacement

File: units/UEL0307/UEL0307_script.lua [hunk 4]
```diff
@@ -87,37 +79,29 @@ UEL0307 = ClassUnit(TShieldLandUnit, ShieldEffectsComponent) {
 
     DisablePointer = function(self)
         self.TargetPointer:SetFireTargetLayerCaps('None') --this disables the stop feature - note that its reset on layer change!
-        self.PointerRestartThread = self:ForkThread(self.PointerRestart)
+        self.PointerRestartThread = self.Trash:Add(ForkThread(self.PointerRestart, self))
```
Comment (Garanas) [inline units/UEL0307/UEL0307_script.lua:82]: Not a valid replacement

File: units/UEL0001/UEL0001_script.lua [hunk 5]
```diff
@@ -208,17 +209,17 @@ UEL0001 = ClassUnit(ACUUnit) {
...
         elseif enh == 'ShieldGeneratorField' then
             self:DestroyShield()
-            self:ForkThread(function()
+            self.Trash:Add(ForkThread(function()
                 WaitTicks(1)
                 self:CreateShield(bp)
                 self:SetEnergyMaintenanceConsumptionOverride(bp.MaintenanceConsumptionPerSecondEnergy or 0)
                 self:SetMaintenanceConsumptionActive()
-            end)
+            end,self))
```
Comment (Hdt80bro) [inline units/UEL0001/UEL0001_script.lua:217]: It doesn't use this parameter, but in general, adding the `self` parameter to the functions to remove an upvalue would be good.

File: units/UEL0301/UEL0301_script.lua [hunk 6]
```diff
@@ -43,14 +42,13 @@ UEL0301 = ClassUnit(CommandUnit) {
 
     OnStopBeingBuilt = function(self, builder, layer)
         CommandUnit.OnStopBeingBuilt(self, builder, layer)
-        -- Block Jammer until Enhancement is built
```
Comment (Hdt80bro) [inline units/UEL0301/UEL0301_script.lua:46]: Don't remove non-trivial comments

File: units/UEL0301/UEL0301_script.lua [hunk 7]
```diff
@@ -120,31 +118,31 @@ UEL0301 = ClassUnit(CommandUnit) {
...
         elseif enh == 'ShieldGeneratorField' then
             self:DestroyShield()
-            self:ForkThread(function()
+            self.Trash:Add(ForkThread(function()
                 WaitTicks(1)
                 self:CreateShield(bp)
                 self:SetEnergyMaintenanceConsumptionOverride(bp.MaintenanceConsumptionPerSecondEnergy or 0)
                 self:SetMaintenanceConsumptionActive()
-            end)
+            end,self))
```
Comment (Hdt80bro) [inline units/UEL0301/UEL0301_script.lua:126]: Same, for the parameter

File: units/UEL0307/UEL0307_script.lua [hunk 8]
```diff
@@ -87,37 +82,32 @@ UEL0307 = ClassUnit(TShieldLandUnit, ShieldEffectsComponent) {
...
     end,
 
     PointerRestart = function(self)
-        --sadly i couldnt find some way of doing this without a thread. dont know where to check if its still assisting other than this.
         while not self.PointerEnabled do
-
-            WaitSeconds(1)
-
-            -- break if we're a gooner
+            WaitTicks(11)
```
Comment (Hdt80bro) [inline units/UEL0307/UEL0307_script.lua:93]: It's good to leave the `WaitSeconds` for readability and intent - this isn't performance critical

File: units/UEL0307/UEL0307_script.lua [hunk 9]
```diff
@@ -87,37 +82,32 @@ UEL0307 = ClassUnit(TShieldLandUnit, ShieldEffectsComponent) {
...
             if IsDestroyed(self) or IsDestroyed(self.TargetPointer) then
                 break
             end
-
-            -- if not gooner, check whether we need to enable our weapon to keep reasonable distance
             if not self:GetGuardedUnit() then
                 self.PointerEnabled = true
-                self.TargetPointer:SetFireTargetLayerCaps(self.TargetLayerCaps[self.Layer]) --this resets the stop feature - note that its reset on layer change!
```
Comment (Hdt80bro) [inline units/UEL0307/UEL0307_script.lua:107]: Please leave in helpful comments

File: units/UES0305/UES0305_script.lua [hunk 10]
```diff
@@ -29,11 +29,11 @@ UES0305 = ClassUnit(TSeaUnit) {
             },
             Type = 'SonarBuoy01',
         },
-    }, 
+    },
 
     CreateIdleEffects = function(self)
         TSeaUnit.CreateIdleEffects(self)
-        self.TimedSonarEffectsThread = self:ForkThread(self.TimedIdleSonarEffects)
+        self.TimedSonarEffectsThread = self.Trash:Add(ForkThread(self.TimedIdleSonarEffects,self))
```
Comment (Hdt80bro) [inline units/UES0305/UES0305_script.lua:36]: This isn't the same

File: units/UES0305/UES0305_script.lua [hunk 11]
```diff
@@ -47,28 +47,30 @@ UES0305 = ClassUnit(TSeaUnit) {
...
-    end,     
+    end,
 
     StartBeingBuiltEffects = function(self, builder, layer)
         self:HideBone(0, true)
         self.BeingBuiltShowBoneTriggered = false
-        if self:GetBlueprint().General.UpgradesFrom ~= builder.UnitId then
-            self.OnBeingBuiltEffectsBag:Add(self:ForkThread(CreateBuildCubeThread, builder, self.OnBeingBuiltEffectsBag))
+        if self.Blueprint.General.UpgradesFrom ~= builder.UnitId then
+            self.OnBeingBuiltEffectsBag:Add(self.Trash:Add(ForkThread(CreateBuildCubeThread, builder, self.OnBeingBuiltEffectsBag, self)))
```
Comment (Hdt80bro) [inline units/UES0305/UES0305_script.lua:73]: This isn't the same

File: units/UES0401/UES0401_script.lua [hunk 12]
```diff
@@ -32,13 +31,15 @@ UES0401 = ClassUnit(AircraftCarrier) {
...
         AircraftCarrier.OnCreate(self)
-        self.OpenAnimManips = {}
-        self.OpenAnimManips[1] = CreateAnimator(self):PlayAnim('/units/ues0401/ues0401_aopen.sca'):SetRate(-1)
+        local openAnim = {}
+        self.OpenAnimManips = openAnim
+        openAnim[1] = CreateAnimator(self):PlayAnim('/units/ues0401/ues0401_aopen.sca'):SetRate(-1)
         for i = 2, 6 do
-            self.OpenAnimManips[i] = CreateAnimator(self):PlayAnim('/units/ues0401/ues0401_aopen0' .. i .. '.sca'):SetRate(-1)
+            openAnim[i] = CreateAnimator(self):PlayAnim('/units/ues0401/ues0401_aopen0' .. i .. '.sca'):
+                SetRate(-1)
```
Comment (Hdt80bro) [inline units/UES0401/UES0401_script.lua:39]: When doing this, put the colon on the same line

File: units/UES0401/UES0401_script.lua [hunk 13]
```diff
@@ -48,10 +49,10 @@ UES0401 = ClassUnit(AircraftCarrier) {
     end,
 
     StartBeingBuiltEffects = function(self, builder, layer)
-        self:SetMesh(self:GetBlueprint().Display.BuildMeshBlueprint, true)
-        if self:GetBlueprint().General.UpgradesFrom ~= builder.UnitId then
+        self:SetMesh(self.Blueprint.Display.BuildMeshBlueprint, true)
+        if self.Blueprint.General.UpgradesFrom ~= builder.UnitId then
             self:HideBone(0, true)
-            self.OnBeingBuiltEffectsBag:Add(self:ForkThread(CreateBuildCubeThread, builder, self.OnBeingBuiltEffectsBag))
+            self.OnBeingBuiltEffectsBag:Add(self.Trash:Add(ForkThread(CreateBuildCubeThread, builder, self.OnBeingBuiltEffectsBag,self)))
```
Comment (Hdt80bro) [inline units/UES0401/UES0401_script.lua:55]: This isn't the same

File: units/UES0401/UES0401_script.lua [hunk 14]
```diff
@@ -74,47 +75,40 @@ UES0401 = ClassUnit(AircraftCarrier) {
...
             self.SinkSlider:SetSpeed(1)
-
             self.SinkSlider:SetGoal(0, difference, 0)
-            WaitSeconds(1)
+            WaitTicks(11)
         end
-
-        self.SinkSlider:SetGoal(0, 0, 0) -- Reset the slider while we are not watching depth
-        WaitFor(self.SinkSlider)-- We have to wait for it to finish before killing the thread or it stops
-
```
Comment (Hdt80bro) [inline units/UES0401/UES0401_script.lua:105]: Leave all of this please

File: units/URA0001/URA0001_script.lua [hunk 15]
```diff
@@ -88,26 +87,25 @@ URA0001 = ClassUnit(CAirUnit) {
...
 
-            local delay = 0.1
+            local delay = 1
             local wait = 0
 
             while wait < 4 do
-                local pos = self:GetPosition()
-                local bpos = self.spawnedBy:GetPosition()
+                local pos = self.Blueprint
+                local bpos = self.spawnedBy.Blueprint
```
Comment (Hdt80bro) [inline units/URA0001/URA0001_script.lua:100]: This isn't the same

File: units/URA0001/URA0001_script.lua [hunk 16]
```diff
@@ -88,26 +87,25 @@ URA0001 = ClassUnit(CAirUnit) {
...
 
                 if VDist2(pos[1], pos[3], bpos[1], bpos[3]) < 1 then
                     break
                 end
 
                 wait = wait + delay
-                WaitSeconds(delay)
+                WaitTicks(delay)
```
Comment (Hdt80bro) [inline units/URA0001/URA0001_script.lua:107]: This isn't the same

File: units/URA0104/URA0104_script.lua [hunk 17]
```diff
@@ -23,10 +17,10 @@ URA0104 = ClassUnit(AirTransport) {
         EMPCannon = ClassWeapon(CEMPAutoCannon) {},
     },
 
-    AirDestructionEffectBones = { 'Left_Exhaust', 'Right_Exhaust', 'Char04', 'Char03', 'Char02', 'Char01',
-                                  'Front_Left_Leg03_B02', 'Front_Right_Leg03_B02', 'Front_Left_Leg01_B02', 'Front_Right_Leg01_B02',
-                                  'Right_AttachPoint01', 'Right_AttachPoint02', 'Right_AttachPoint03', 'Right_AttachPoint04',
-                                  'Left_AttachPoint01', 'Left_AttachPoint02', 'Left_AttachPoint03', 'Left_AttachPoint04', },
+    AirDestructionEffectBones = { 'Left_Exhaust', 'RighF_Exhaust', 'Char04', 'Char03', 'Char02', 'Char01',
```
Comment (Hdt80bro) [inline units/URA0104/URA0104_script.lua:20]: Is the bone really misspelled?

File: units/URA0104/URA0104_script.lua [hunk 18]
```diff
@@ -35,48 +29,49 @@ URA0104 = ClassUnit(AirTransport) {
...
     end,
 
     AirDestructionEffectsThread = function(self)
         local numExplosions = math.floor(table.getn(self.AirDestructionEffectBones) * 0.5)
         for i = 0, numExplosions do
-            explosion.CreateDefaultHitExplosionAtBone(self, self.AirDestructionEffectBones[util.GetRandomInt(1, numExplosions)], 0.5)
-            WaitSeconds(util.GetRandomFloat(0.2, 0.9))
+            explosion.CreateDefaultHitExplosionAtBone(self,
+                self.AirDestructionEffectBones[util.GetRandomInt(1, numExplosions)], 0.5)
+            WaitTicks(util.GetRandomFloat(2, 9))
```
Comment (Hdt80bro) [inline units/URA0104/URA0104_script.lua:69]: This isn't quite the same

File: units/URA0401/URA0401_script.lua [hunk 19]
```diff
@@ -31,55 +31,56 @@ URA0401 = ClassUnit(CAirUnit) {
...
-        if self.ThrustExhaustTT1 == nil then 
+        if self.ThrustExhaustTT1 == nil then
             if self.MovementAmbientExhaustEffectsBag then
-                fxutil.CleanupEffectBag(self,'MovementAmbientExhaustEffectsBag')
+                fxutil.CleanupEffectBag(self, 'MovementAmbientExhaustEffectsBag')
             else
                 self.MovementAmbientExhaustEffectsBag = {}
             end
-            self.ThrustExhaustTT1 = self:ForkThread(self.MovementAmbientExhaustThread)
+            self.ThrustExhaustTT1 = self.Trash:Add(ForkThread(self.MovementAmbientExhaustThread,self))
```
Comment (Hdt80bro) [inline units/URA0401/URA0401_script.lua:53]: This isn't the same

**Review by Hdt80bro — CHANGES_REQUESTED** [review]

**Pull request closed without merging**